app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -82,13 +67,10 @@ def create
       param_group :compute_resource
 
       def update
-        datacenter = change_datacenter_to_uuid(compute_resource_params[:datacenter])
```

Comment:
Datacenter should be still available for vmware (at least according to the comment above). Why did you remove it from here?

Reply:
In `change_datacenter_to_uuid` is condition `@compute_resource.respond_to?(:get_datacenter_uuid)`

VMware doesn't have this method, so it's safe to remove it.